code/game/area/areas.dm

```diff
@@ -33,6 +33,8 @@
 
 	var/mood_bonus = 0 //Mood for being here
 	var/mood_message = "<span class='nicegreen'>This area is pretty nice!\n</span>" //Mood message for being here, only shows up if mood_bonus != 0
+	var/list/mood_job_allowed = list()  // if defined, restricts what jobs get this buff using JOB_NAME defines (-candycane/etherware)
+	var/mood_reverse = FALSE  // if TRUE, mood_job_allowed is a list of jobs not affected
```

Comment:
Bad variable name, my first thought is that this reverses the mood effect rather than the job check